app/controllers/news_controller.rb

```diff
@@ -37,13 +47,33 @@ def show
 
 
   private
+  # An array of sections the user is filtering by, loaded by params
+  # with section ids
+  def set_filters
+    @filters = []
+
+    [:ministers, :departments].each do |filter_type|
+      if params[filter_type].present?
+        id_list = params[filter_type].map(&:to_i)
+
+        id_list.reject! do |item|
+          !Section.exists?(item)
```

Comment:
This will incur a DB round trip every run through the loop. Instead, preload all the section IDs into an array and check that.